### PR TITLE
Release/0.2.0

### PR DESCRIPTION
**Pull request by DavidSubiros**
### What

Release branch for `Retrieve information from Cantabular`
Also contains DockerFile.local

Trello card: https://trello.com/c/MVhVQLlm/5149-retrieve-information-from-cantabular-apidp-cantabular-import-dimension-options

### How to review

Check only expected commits are present in this PR

### Who can review

Anyone

### CODE REVIEW

**Review by redhug1 — COMMENTED** [review]
Just looking at your latest PR and trying:
make test-component
and as the lines scrolled by …
it paused and i copied down the following:
warning: no packages being tested depend on matches for pattern github.com/ONSdigital/dp-import-cantabular-dataset/...

and then as it ran further on, it ended up with an error:
dp-import-cantabular-dimension-options_1  |   Scenario: Posting and checking a response      # features/helloworld.feature:5
dp-import-cantabular-dimension-options_1  |     When these hello events are consumed:        # steps.go:30 -> *Component
dp-import-cantabular-dimension-options_1  |       | RecipientName |
dp-import-cantabular-dimension-options_1  |       | Tim           |
dp-import-cantabular-dimension-options_1  |     Then I should receive a hello-world response # steps.go:25 -> *Component
dp-import-cantabular-dimension-options_1  |
dp-import-cantabular-dimension-options_1  | 1 scenarios (1 passed)
dp-import-cantabular-dimension-options_1  | 2 steps (2 passed)
dp-import-cantabular-dimension-options_1  | 3.968989041s
dp-import-cantabular-dimension-options_1  | PASS
dp-import-cantabular-dimension-options_1  | coverage: [no statements]
dp-import-cantabular-dimension-options_1  | ok  	github.com/ONSdigital/dp-import-cantabular-dimension-options	4.005s
dp-import-cantabular-dimension-options_1 exited with code 0
Aborting on container exit...
[+] Running 3/3
 ⠿ Container dp-import-cantabular-dimension-options-features_zookeeper_1                               Stopped                                                                                   10.2s
 ⠿ Container dp-import-cantabular-dimension-options-features_dp-import-cantabular-dimension-options_1  Stopped                                                                                    0.0s
 ⠿ Container dp-import-cantabular-dimension-options-features_kafka_1                                   Stopped                                                                                    3.4s
ERRO[0165] 0

Where it says Aborting on container exit … is there any chance you can output some message at this point to say to ignore the final ERRO[0165] 0  … if that is an OK sort of error ?

File: ci/scripts/lint.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash -eux
+
+cwd=$(pwd)
+
+pushd $cwd/dp-import-cantabular-dimension-options
+  make lint
+popd
```
Comment (redhug1) [inline ci/scripts/lint.sh:7]: This line needs a new line

File: reflex [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Watch all files ending in .go, excluding files in `.go` or `cmd` directory
+# or files ending in `_test.go` & run `make debug-run` when any matching file
+# is changed
+-s -r \.go$ -R ^\.go/ -R ^cmd/ -R _test\.go$ make debug-run
```
Comment (redhug1) [inline reflex:4]: needs a new line added

File: service/interfaces.go [hunk 3]
```diff
@@ -4,11 +4,15 @@ import (
...
 	"net/http"
 
+	"github.com/ONSdigital/dp-api-clients-go/cantabular"
+	"github.com/ONSdigital/dp-api-clients-go/dataset"
 	"github.com/ONSdigital/dp-healthcheck/healthcheck"
 )
 
 //go:generate moq -out mock/server.go -pkg mock . HTTPServer
 //go:generate moq -out mock/healthCheck.go -pkg mock . HealthChecker
+//go:generate moq -out mock/cantabular-client.go -pkg mock . CantabularClient
```
Comment (redhug1) [inline service/interfaces.go:14]: Hmm commented out lines ?

**Review by redhug1 — CHANGES_REQUESTED** [review]
service_test.go
lines 247, 262 : Error string should not be capitalized or end with punctuation mark

consumer_test.go
line 19 : Error string should not be capitalized or end with punctuation mark
line 31: Unused variable 'kafkaStubConsumer'

callback.go
line 22: Unused function 'logData'

cmd/producer/main.go :
line 24: Variable 'config' collides with imported package name

cmd/consumer/main.go :
line 24: Variable 'config' collides with imported package name
line 66 : Unhandled error

-=-=-
The file 'reflex' in the repo's root has a comment that explains what it does.
Please can you add (with joshua's assistance if needs be) additional comment(s) that help
to educate everyone else where this is used and what its purpose is where it is used.
Thanks very much.

File: service/interfaces.go (hunk 3 above)
Comment (DavidSubiros) [reply service/interfaces.go:14]: Yes, this is to let `go generate` know that it has to create a mock for the CantabularClient interface

**Comment by DavidSubiros** [comment]
> Just looking at your latest PR and trying:
> make test-component
> and as the lines scrolled by …
> it paused and i copied down the following:
> warning: no packages being tested depend on matches for pattern github.com/ONSdigital/dp-import-cantabular-dataset/...
> 
> and then as it ran further on, it ended up with an error:
> dp-import-cantabular-dimension-options_1 | Scenario: Posting and checking a response # features/helloworld.feature:5
> dp-import-cantabular-dimension-options_1 | When these hello events are consumed: # steps.go:30 -> *Component
> dp-import-cantabular-dimension-options_1 | | RecipientName |
> dp-import-cantabular-dimension-options_1 | | Tim |
> dp-import-cantabular-dimension-options_1 | Then I should receive a hello-world response # steps.go:25 -> *Component
> dp-import-cantabular-dimension-options_1 |
> dp-import-cantabular-dimension-options_1 | 1 scenarios (1 passed)
> dp-import-cantabular-dimension-options_1 | 2 steps (2 passed)
> dp-import-cantabular-dimension-options_1 | 3.968989041s
> dp-import-cantabular-dimension-options_1 | PASS
> dp-import-cantabular-dimension-options_1 | coverage: [no statements]
> dp-import-cantabular-dimension-options_1 | ok github.com/ONSdigital/dp-import-cantabular-dimension-options 4.005s
> dp-import-cantabular-dimension-options_1 exited with code 0
> Aborting on container exit...
> [+] Running 3/3
> ⠿ Container dp-import-cantabular-dimension-options-features_zookeeper_1 Stopped 10.2s
> ⠿ Container dp-import-cantabular-dimension-options-features_dp-import-cantabular-dimension-options_1 Stopped 0.0s
> ⠿ Container dp-import-cantabular-dimension-options-features_kafka_1 Stopped 3.4s
> ERRO[0165] 0
> 
> Where it says Aborting on container exit … is there any chance you can output some message at this point to say to ignore the final ERRO[0165] 0 … if that is an OK sort of error ?

i guess this one corresponds to `feature/component-tests`, I will try to put more logs there (and check that the CI shows them?) - it's still in progress

**Review by redhug1 — APPROVED** [review]

**Pull request merged**